src/main/java/seedu/address/model/predicate/StudentHasPaidPredicate.java

```diff
@@ -34,8 +35,8 @@ public boolean equals(Object other) {
             return false;
         }
 
-        StudentHasPaidPredicate studentHasPaidPredicate = (StudentHasPaidPredicate) other;
-        return this.paymentUpToDate == studentHasPaidPredicate.paymentUpToDate;
+        StudentHasPaidPredicate otherStudentHasPaidPredicate = (StudentHasPaidPredicate) other;
```

Comment:
Here could use stream too to make this class consistent with the rest
eg. anyMatch(overdueAmount -> Integer.parseInt(overdueAmount) > 0)